core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2106,6 +2106,11 @@ public static boolean isInf(double b0) {
     return Double.isInfinite(b0);
   }
 
+  /** SQL <code>IS_INF</code> operator applied to float values. */
```

Comment:
do the tests pass if you remove the `isInf(float)` and isNaN(float)` methods? I suspect they will. If so, remove the methods. less is more.

Reply:
Done.

Reply:
This was actually not right. The tests fail because if you remove the `float` methods, they use the BigDecimal functions which do not recognize `Infinity` and `NaN`

Reply:
Could you add a test case for float value? IMO this may go to the function applied to double value, if we remove the function applied to float value. I am not trying it, so please let me know if I'm wrong.

Reply:
Done